examples/spa/associative_memory.ipynb

```diff
@@ -0,0 +1,462 @@
+      "\n",
+      "sim = nengo.Simulator(model3)\n",
+      "sim.run(0.2)\n",
+      "t = sim.trange()\n",
+      "\n",
+      "input_net = sim.data[input_probe]\n",
+      "output_net = sim.data[output_probe]\n",
+      "\n",
+      "plot_similarities(input_net, output_net, vocab, top=1.1)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "We see that the input is similar to `CHERRY` and `APPLE` and somewhat similar to `APRICOT`. As we have specified the threshold of `0.4` the resulting output vector will be similar to `APPLE` and `CHERRY` as their similarity with the input vector is greater than the threshold."
+     ]
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Step 3: Hetero-associative relationships\n",
+      "\n",
+      "To model a variety of interesting memory phenomena, it is often useful to store relationships between different sets of patterns.\n",
+      "For example, to simulate number counting from 1 to 5, the memory needs to store the relationships between patterns representing numbers:\n",
+      "$1\\rightarrow2, 2\\rightarrow3, 3\\rightarrow4, 4\\rightarrow5$\n",
+      "\n",
+      "In this example we show how to use `spa.AssociativeMemory` to this task. In order to achieve the number counting we will split the task into two parts:\n",
+      "\n",
+      "1. Present a number at the input and recall a number greater by one (e.g. for `1` recall `2`, for `2` recall `3` etc)\n",
+      "    \n",
+      "2. Feed back the output of the associative memory to its input\n",
+      "\n",
+      "    \n",
+      "As in the previous example, we start by defining a vocabulary that stores semantic pointes representing five numbers. This will be used as the input and as the output vocabulary. In case of hetero-associative memories the input and the output vocabulary can differ. Because we want to achieve a hetero-associative mapping, we need to specify which input patterns map to which output patterns. The desired mapping in specified by providing `input_keys` and `output_keys`:"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "dim = 16\n",
+      "vocab_numbers = spa.Vocabulary(dimensions=dim)\n",
+      "\n",
+      "# a quicker way to add words to a vocabulary\n",
+      "vocab_numbers.parse('ONE + TWO + THREE + FOUR + FIVE')\n",
+      "\n",
+      "# from patterns\n",
+      "input_keys = ['ONE', 'TWO', 'THREE', 'FOUR']\n",
+      "\n",
+      "# to patterns\n",
+      "output_keys = ['TWO', 'THREE', 'FOUR', 'FIVE']"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "Now we introduce a Nengo's feature that allows us to pass a function as the input to a `Node`. \n",
```

Comment:
I think, users should already be familiar with passing functions to nodes at this point.